phylogeny/prototypeSelection.py

```diff
@@ -0,0 +1,99 @@
+"""
+Prototype selection
+
+Given a set of n elements (S) and their pairwise distances in terms of a
+skbio distance matrix (DM) and a given number (k << n), find the sub-set (s)
+consisting of exactly k elements, such that s best represents the full set S.
+
+Here, we define "best represents" as maximizing the sum of distances between
+all points, i.e. sum {over a,b in S} DM[a,b]. This is our objective function.
+
+This problem is known to be NP-hard [1], thus we need to resort to heuristics.
+In the future, this module will implement several heuristics, whose quality can
+be measured by the objective function for each problem instance, since there is
+no global winner.
+For completeness, the exact but exponential algorithm is implemented, too.
+  "prototypeSelection_exhaustive"
+
+[1] Gamez, J. Esteban, François Modave, and Olga Kosheleva.
+    "Selecting the most representative sample is NP-hard:
+     Need for expert (fuzzy) knowledge."
+    Fuzzy Systems, 2008. FUZZ-IEEE 2008.
+"""
+
+from skbio.stats.distance import DistanceMatrix
```

Comment:
Minor, but the numpy standards and/or PEP8 defines a specific order in which the imports should be done:
```
standard library imports 
blank line
3rd party tools imports
blank line
local project imports
```

Following this rule, your imports should look like:

```python
from typing import Sequence, Tuple
import itertools  # can you use from itertools import XXX rather than importing the entire module?

import numpy as np
import scipy as sp
from skbio.stats.distance import DistanceMatrix
```